index.js

```diff
@@ -11,25 +11,47 @@ let loggerGen = require("heimdalljs-logger");
 let cleanUpResolvedPath = cleanUpPath.cleanUpResolvedPath;
 let isResolved = cleanUpPath.isResolved;
 
+// 5 seconds. We'll try to cache for a bit of time, to avoid scanning same files
+// in a single rebuild.
+const CACHE_TTL = 5000;
+
+const GLOBAL_CACHE = new Map();
+
 class DirectoryIndex extends Map {}
 
 class Directory extends String {
   constructor(p, cleanedUp) {
     if (!cleanedUp) p = cleanUpPath(p);
     super(p);
+
+    this._isNodeModulesPackage = p.includes('node_modules');
   }
 
   getIndexSync() {
-    if (this._index != null) return this._index;
+    if (this._index != null) {
+      if (this._isNodeModulesPackage) {
+        return this._index;
+      }
+
+      const now = Date.now();
+      const isCacheValid = (this._cacheBuildAt + CACHE_TTL) > now;
```

Comment:
how does this affect the dev-time reload of linked packages that are maybe `ctrl+s`'d rapidly?
Like, does this mean that when working on a library, we can only receive updates at quickest, every 5s?

Reply:
For now yes, it was more of an experiment and solution to our problem.. with this - we have issues in tests for the merge-trees, so need to figure out some other solution.. but to say, it does add a big speed boost, so caching is a must.. need to think how to pass it probably from the consuming packages.